src/couchbase-server.py

```diff
@@ -0,0 +1,57 @@
+import re
+from bs4 import BeautifulSoup
+from common import endoflife
+from datetime import datetime
+
+"""Fetch versions with their dates from docs.couchbase.com.
+
+Versions are fetched from the release notes of each minor version. Date is not
+accurate (only month and year are provided) but it's already better than the
+previous implementation which use dates of the images on Docker Hub. Because
+of that lack of precision, all dates are set to the 15th of the month.
+
+Notes:
+- documentation for oldest versions is not available anymore on
+  docs.couchbase.com, so this script is using the Wayback Machine to fetch them.
+- dates are not available for some versions, those had to be set manually.
+"""
+
+PRODUCT = "couchbase-server"
+REGEX = r"^Release (?P<version>\d+\.\d+(\.\d+)?) \((?P<date>.+)\)$"
+URLS = [
+    "https://web.archive.org/web/20230519160357/https://docs.couchbase.com/server/",
+    "https://docs.couchbase.com/server",
+]
+FIXED_VERSIONS = {
+    "6.0.0": "2018-10-31",  # https://www.couchbase.com/blog/announcing-couchbase-6-0/
+    "6.0.1": "2019-02-15",  # https://web.archive.org/web/20190307191211/https://docs.couchbase.com/server/6.0/release-notes/relnotes.html
+    "7.2.0": "2023-06-01",  # https://www.couchbase.com/blog/couchbase-capella-spring-release-72/
+}
```

Comment:
Not for now, but thoughts about having a separate `patches` directory that holds corrections?

Reply:
Not yet, I am thinking it makes more sense to keep it in the same script at the moment. But will think about it.